torch/_inductor/codegen/cpp.py

```diff
@@ -2241,7 +2149,16 @@ def _legalize_bf16(loop_body: ir.LoopBody):
         for _node in nodes:
             assert isinstance(_node, SchedulerNode)
             node: SchedulerNode = _node
-            if isinstance(node._body, ir.LoopBody):
+
+            def is_memory_copy_schedulernode(node: SchedulerNode):
+                op_counts = node.read_writes.op_counts
+                return (
+                    len(op_counts) == 2 and "load" in op_counts and "store" in op_counts
+                )
+
+            should_add_to_dtype = not is_memory_copy_schedulernode(node)
+
+            if should_add_to_dtype and isinstance(node._body, ir.LoopBody):
```

Comment:
I guess we can do assertion on `isinstance(node._body, ir.LoopBody)` instead of conditional check?